crossdock/server/server.py

```diff
@@ -77,25 +83,41 @@ def head(self):
 def make_app(server):
     return tornado.web.Application(
         [
-            (r"/", MainHandler),
-            (r"/start_trace", MainHandler, (dict(server=server, method=Server.start_trace))),
-            (r"/join_trace", MainHandler, (dict(server=server, method=Server.join_trace))),
+            (r'/', MainHandler),
+            (r'/start_trace', MainHandler, (dict(server=server, method=Server.start_trace))),
+            (r'/join_trace', MainHandler, (dict(server=server, method=Server.join_trace))),
         ], debug=True)
 
 
-# TChannel handlers
-def tchannelNotImplResponse():
-    return TraceResponse(notImplementedError="python <=> tchannel not implemented")
+def make_tchannel(port):
+    global tchannel
+    tchannel = TChannel('python', hostport='localhost:%d' % port, trace=True)
+
+    service = get_thrift_service(service_name='python')
+
+    @tchannel.thrift.register(service.TracedService, method='joinTrace')
+    @tornado.gen.coroutine
+    def join_trace(request):
+        join_trace_request = request.body.request or None
+        response = yield handle_downstream_request(join_trace_request.downstream)
+        raise tornado.gen.Return(response)
+
+    @tornado.gen.coroutine
+    def handle_downstream_request(downstream):
```

Comment:
nit: this would be better named `prepare_response`, because it does not always do downstream